homeassistant/components/netatmo/button.py

```diff
@@ -0,0 +1,74 @@
+"""Support for Netatmo/Bubendorff button."""
+
+from __future__ import annotations
+
+import logging
+
+from pyatmo import modules as NaModules
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_URL_CONTROL, NETATMO_CREATE_BUTTON
+from .data_handler import HOME, SIGNAL_NAME, NetatmoDevice
+from .entity import NetatmoModuleEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Netatmo button platform."""
+
+    @callback
+    def _create_entity(netatmo_device: NetatmoDevice) -> None:
+        entity = NetatmoCoverPreferredPositionButton(netatmo_device)
+        _LOGGER.debug("Adding button %s", entity)
+        async_add_entities([entity])
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, NETATMO_CREATE_BUTTON, _create_entity)
+    )
+
+
+class NetatmoCoverPreferredPositionButton(NetatmoModuleEntity, ButtonEntity):
+    """Representation of a Netatmo cover preferred position button device."""
+
+    _attr_configuration_url = CONF_URL_CONTROL
+    device: NaModules.Shutter
+
+    def __init__(self, netatmo_device: NetatmoDevice) -> None:
+        """Initialize the Netatmo device."""
+        super().__init__(netatmo_device)
+
+        self._attr_translation_key = "preferred_position"
+        self._attr_entity_registry_enabled_default = False
```

Comment:
these can be defined as class attributes, like `_attr_configuration_url, which makes the class a bit cleaner